src/inmanta/plugins.py

```diff
@@ -172,6 +177,28 @@ def get_functions(cls) -> Dict[str, "Type[Plugin]"]:
         """
         return dict(cls.__functions)
 
+    @classmethod
+    def get_originals(cls) -> Dict[str, str]:
+        """
+        Get the original name of all the functions with the associated function name
+        """
+        return dict(cls.__function_name_mapping)
+
+    @classmethod
+    def deprecate_function(cls, fnc: Callable, replaced_by: Optional[str] = None) -> None:
+        name = fnc.__name__
+        ns_parts = str(fnc.__module__).split(".")
+        ns_parts.append(name)
+        full_name = "::".join(ns_parts[1:])
+        if hasattr(fnc, "plugin_name"):
+            cls.__functions[fnc.plugin_name].deprecated = True
+            cls.__functions[fnc.plugin_name].replaced_by = replaced_by
+        else:
+            raise Exception(
+                f"Can not deprecate plugin '{full_name}': The '@deprecated' decorator should be used in combination with the "
```

Comment:
I don't think it makes sense to use the inmanta plugin name in this error message because it's not a plugin, therefore it doesn't have an inmanta name, only a Python name.